### PR TITLE
[PD] Fix of requests occasionally missing issue in async transfer (SW-234952)

### PR DESCRIPTION
**Pull request by hlin99**

### CODE REVIEW

**Review by jikunshang — APPROVED** [review]

File: vllm/worker/hpu_model_runner.py [hunk 1]
```diff
@@ -2946,14 +2946,19 @@ def async_recv_kv_caches(model, model_input, attn_metadata,
...
                         k_v_head_size = 576
                         bypass_model_exec = True
                         htorch.core.mark_step()
                         for idx, slen in enumerate(seq_lens):
                             if slen == 1:
+                                dummy_hidden = torch.zeros(HIDDEN_SHAPE, device="hpu")
+                                dummy_kv = torch.zeros(KV_SHAPE, device="hpu")
+
                                 hidden_states_list.append(
-                                    hidden_states_list[0])
```
Comment (jerrychenhf) [inline vllm/worker/hpu_model_runner.py:2956]: Will there be a case that the padding sequence (slen == 1) appears at the first sequence in a batch?
If there is no such case, what's the problem using hidden_states_list[0] to reuse other's hidden state since it is dummy?

Comment (jerrychenhf) [reply vllm/worker/hpu_model_runner.py:2956]: I see. So we need a dummy hidden states.  But can we remove unused dummy_kv ?
KV_SHAPE = (61, 128, 1, 576)
dummy_kv = torch.zeros(KV_SHAPE, device="hpu")

**Comment by jerrychenhf** [comment]
LGTM. Thank!
